### ISSUE TITLE
spyOn(obj, "method").withArgs("a", "b").andReturn("c")

### ISSUE THREAD

**Issue by Peeja** [jasmine/jasmine#94]
RSpec stubs let you specify return values for specific invocations.  It would be nice if `spyOn` could do the same thing.  The current workaround is to use `andCallFake` with a big conditional:

```
spyOn(obj, "method").andCallFake(function(a, b) {
  if (a == "a" && b == "b") {
    return "c";
  } else {
    throw "Unexpected arguments for method()"
  }
}
```

which is a lot of boilerplate once you use it a few times.

**Comment by johnjcorcoran** [comment]
This seems like a fairly fundamental part of a testing framework, so I'm surprised that it's not explictly supported. Is there a better suggested alternative than using ``andCallFake`` and conditionals? Doing that seems sloppy and error-prone.

Not checking the arguments used in function calls seems even worse, as breaking code changes can be made without any tests failing.

**Comment by sarod** [comment]
@infews this feature would be very useful. Could you give us a bit more insight on why this was closed.

I can't find anything on the jasmine mailing list except a [similar request from 2014](https://groups.google.com/d/msg/jasmine-js/A3FY2oOfops/3U9IWwVuLrwJ) that is still without response. Are you talking about another mailing list?

What was the conclusion of the "good discussion"? Did you conclude it was useless or not doable?

**Comment by slackersoft** [comment]
The current plan to support this kind of functionality is to see how far we can get if we allow users to specify custom SpyStrategies. We haven't gotten the time to implement this feature yet, but would be happy to review a pull request to implement custom SpyStrategies.

**Comment by draconteus** [comment]
Is there any progress about this? Does anybody working on making it real?

**Comment by devkeydet** [comment]
I would like to know too.  Currently, I use sinon stubs for this:

sinon.stub(obj, "functionName").withArgs("matchingValue").returns("somevalue");

This is really the only thing I use sinon for instead of Jasmine spies.  So it would be great to eliminate the need for sinon.

**Comment by pjmalandrino** [comment]
Now you can do :
spyOn(resource, 'method').withArgs('argument');

Available since 3.0.0
https://jasmine.github.io/api/edge/Spy

**Comment by jeetporia** [comment]
npm i -D jasmine@latest jasmine-core@latest @types/jasmine@latest
